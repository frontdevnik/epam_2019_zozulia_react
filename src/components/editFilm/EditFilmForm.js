import React from 'react'
import { reduxForm, Field } from 'redux-form';
import withTranslation from '../../hocs/withTranslation';
import styles from './editFilm.module.scss';

function EditFilmForm(props) {
  const { handleSubmit, goBack, editFilm_title, editFilm_img, editFilm_director, editFilm_genres, editFilm_description, editFilm_submit, editFilm_goBack } = props;
  return (
    <form className={styles.editForm} onSubmit={handleSubmit}>
      <label>
        {editFilm_title}:
        <Field
          name="title"
          component="input"
          type="text"
        />
      </label>
      <label>
        {editFilm_img}:
        <Field
          name="posterUrl"
          type="text"
          component="input"
        />
      </label>
      <label>
        {editFilm_director}:
        <Field
          name="director"
          type="text"
          component="input"
        />
      </label>
      <label>
        {editFilm_genres}:
        <Field
          name="genres"
          type="text"
          component="input"
        />
      </label>
      <label>
        {editFilm_description}:
        <Field
          name="description"
          type="text"
          component="textarea"
        />
      </label>
      <div className={styles.buttons}>
        <button>{editFilm_submit}</button>
        <button type='button' onClick={goBack}>{editFilm_goBack}</button>
      </div>
    </form>
  )
}

const withReduxForm = reduxForm({
  form: 'editFilm-form',
});

const withTranslationWords = withTranslation(['editFilm_title', 'editFilm_img', 'editFilm_director', 'editFilm_genres', 'editFilm_description', 'editFilm_submit', 'editFilm_goBack']);

export default withTranslationWords(withReduxForm(EditFilmForm));
