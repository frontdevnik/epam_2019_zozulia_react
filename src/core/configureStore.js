import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import callApi from './callApi';
import rootReducer from './rootReducer';

// Redux dev tools
let devTools = (f) => f;

const enableReduxDevtools = process.browser
  && process.env.NODE_ENV !== 'production'
  && window.__REDUX_DEVTOOLS_EXTENSION__;

if (enableReduxDevtools) {
  devTools = window.__REDUX_DEVTOOLS_EXTENSION__();
}

export const configureStore = (initialState = {}) => createStore(
  rootReducer,
  initialState,
  compose(
    applyMiddleware(thunk.withExtraArgument(callApi)),
    devTools,
  ),
);
