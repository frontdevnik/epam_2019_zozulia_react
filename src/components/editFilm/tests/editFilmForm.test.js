import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import EditFilmForm from '../EditFilmForm';
import styles from '../editFilm.module.scss';

describe('<EditFilmForm />', () => {
  let wrapper;
  let store;
  const mockStore = configureStore();
  const initialState = {
    actorReducer: {
      actor: null,
      loading: true,
      error: null,
    },
  };

  const props = {
    editFilm_title: 'Title', 
    editFilm_img: 'Img url', 
    editFilm_director: 'Director', 
    editFilm_genres: 'Genres', 
    editFilm_description: 'Description', 
    editFilm_submit: jest.fn(),
    goBack: jest.fn(),
    editFilm_goBack: 'Go back',
  }

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = shallow(<Provider store={store}><EditFilmForm {...props} /></Provider>);
  });

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
