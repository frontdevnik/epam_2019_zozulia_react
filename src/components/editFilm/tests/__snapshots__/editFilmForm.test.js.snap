// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditFilmForm /> should render correctly 1`] = `"<form class=\\"editForm\\"><label>Title:<input type=\\"text\\" name=\\"title\\" value=\\"\\"/></label><label>Img url:<input type=\\"text\\" name=\\"posterUrl\\" value=\\"\\"/></label><label>Director:<input type=\\"text\\" name=\\"director\\" value=\\"\\"/></label><label>Genres:<input type=\\"text\\" name=\\"genres\\" value=\\"\\"/></label><label>Description:<textarea name=\\"description\\" type=\\"text\\"></textarea></label><div class=\\"buttons\\"><button>Submit</button><button type=\\"button\\">Go back</button></div></form>"`;

exports[`<EditFilmForm /> should render correctly 2`] = `"<form class=\\"editForm\\"><label>Title:<input type=\\"text\\" name=\\"title\\" value=\\"\\"/></label><label>Img url:<input type=\\"text\\" name=\\"posterUrl\\" value=\\"\\"/></label><label>Director:<input type=\\"text\\" name=\\"director\\" value=\\"\\"/></label><label>Genres:<input type=\\"text\\" name=\\"genres\\" value=\\"\\"/></label><label>Description:<textarea name=\\"description\\" type=\\"text\\"></textarea></label><div class=\\"buttons\\"><button>Submit</button><button type=\\"button\\">Go back</button></div></form>"`;
